static/js/blueslip.js

```diff
@@ -358,40 +373,58 @@ function build_arg_list(msg, more_info) {
     return args;
 }
 
+ // Specify which log messages should be allowed or blocked.
+ // Parameters: getLogRulesChecker(LOG_ALL, ALLOWED, BLOCKED)
+var shouldWeLog = getLogRulesChecker(
+                    true,
+                    [],
+                    ["Narrowed","Unnarrowed"]
+                );
+
 exports.debug = function blueslip_debug (msg, more_info) {
-    var args = build_arg_list(msg, more_info);
-    logger.debug.apply(logger, args);
+    if (shouldWeLog(msg)) {
+        var args = build_arg_list(msg, more_info);
+        logger.debug.apply(logger, args);
+    }
 };
 
 exports.log = function blueslip_log (msg, more_info) {
-    var args = build_arg_list(msg, more_info);
-    logger.log.apply(logger, args);
+    if (shouldWeLog(msg)) {
+        var args = build_arg_list(msg, more_info);
+        logger.log.apply(logger, args);
+    }
 };
 
 exports.info = function blueslip_info (msg, more_info) {
-    var args = build_arg_list(msg, more_info);
-    logger.info.apply(logger, args);
+    if (shouldWeLog(msg)) {
+        var args = build_arg_list(msg, more_info);
+        logger.info.apply(logger, args);
+    }
 };
 
 exports.warn = function blueslip_warn (msg, more_info) {
-    var args = build_arg_list(msg, more_info);
-    logger.warn.apply(logger, args);
-    if (page_params.debug_mode) {
-        console.trace();
+    if (shouldWeLog(msg)) {
+        var args = build_arg_list(msg, more_info);
+        logger.warn.apply(logger, args);
+        if (page_params.debug_mode) {
+            console.trace();
+        }
     }
 };
 
 exports.error = function blueslip_error (msg, more_info, stack) {
-    if (page_params.debug_mode) {
-        console.log(stack);
-        throw new BlueslipError(msg, more_info);
-    } else {
-        if (stack === undefined) {
-            stack = Error().stack;
+    if (shouldWeLog(msg)) {
+        if (page_params.debug_mode) {
+            console.log(stack);
```

Comment:
I think here even if the logging is disabled, we would still like to report the error.